Revert "Use in-memory ascii-armored keys instead of using the gpg agent"

kneth opened the pull request:
This reverts commit 90799690dde14f0d27614cba34f8783b11a01bae and switches to sonatype tokens.

## Summary of the changes / Why this is an improvement


## Checklist

 - [ ] Link to issue this PR refers to (if applicable): Fixes #???

seut reviewed (commented):
great!
Could you please be a bit more verbose with your commit messages (at least the once which should be merged)?
E.g. the last one could be:
```
Configure sonatype username/password access-token
Sonatype now requires a access token username/password in order to
publish artifacts

Cherry-picked from https://github.com/crate/jmx_exporter/commit/b287ea5
```

Pointing to some file without any more context may not be really helpful. More concrete message will help us in future to figure out what happened and why.
Thank you!

kneth commented:
@seut 

> Could you please be a bit more verbose with your commit messages

Oh, I can only agree. I have squashed the commits and changed the commit message.
